### PR TITLE
Document Related JS Packages (Back Doors)

### PR DESCRIPTION
**Pull request by harrysolovay**
*Issue #, if available:* #2359 

*Description of changes:*

Added descriptions of related JavaScript packages / back doors for interacting with services (in case the users' use case is unsupported).

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/lib/auth/fragments/js/advanced.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+## Back Doors
```
Comment (jpignata) [inline docs/lib/auth/fragments/js/advanced.md:1]: Do we use this language elsewhere? We typically refer to this kind of facility as an "escape hatch" so I'm curious about why you chose this nomenclature.

File: docs/lib/auth/fragments/js/advanced.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+
+Depending on your use case, you may want to utilize either [`amazon-cognito-identity-js`](https://github.com/aws-amplify/amplify-js/tree/main/packages/amazon-cognito-identity-js) (the User Pool client, which underlies Amplify Auth), [`aws-sdk-js`](https://github.com/aws/aws-sdk-js), or [`aws-sdk-v3-js`](https://github.com/aws/aws-sdk-js-v3) directly.
+
+### `amazon-cognito-identity-js`
+
+The [`amazon-cognito-identity-js`](https://github.com/aws-amplify/amplify-js/tree/main/packages/amazon-cognito-identity-js) package is responsible for forming AWS requests for common User Pool actions, while abstracting away the need to manually persist sessions and tokens between those requests. It also computes [SRP](https://en.wikipedia.org/wiki/Secure_Remote_Password_protocol) values to secure your authentication and challenge requests. This package is maintained with [the `amplify-js` repository](https://github.com/aws-amplify/amplify-js) (copied from [its original repository](https://github.com/amazon-archives/amazon-cognito-identity-js), which has since been deprecated and moved into [`amazon-archives`](https://github.com/amazon-archives)).
```
Comment (jpignata) [inline docs/lib/auth/fragments/js/advanced.md:7]: The provenance of the code in terms of repo doesn't seem relevant in a documentation context. Is there documentation about this underlying code you can link to instead?

File: docs/lib/auth/fragments/js/advanced.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+
+### `aws-sdk`
+
+The current version of [`aws-sdk-js`](https://github.com/aws/aws-sdk-js) enables service-level requests (no abstraction). This SDK **is NOT modularized**, meaning that it can dramatically increase the weight of your builds. Alternatively, you can use [`aws-sdk-js-v3`](https://github.com/aws/aws-sdk-js-v3). While this package is currently in Gamma, it is relatively safe. In fact, it is used in production for this very site and is currently running in your browser. V3 enables you to import specific commands, and tree shake those left unused.
```
Comment (jpignata) [inline docs/lib/auth/fragments/js/advanced.md:11]: "Gamma" is not typically a customer-facing term. I notice in their readme they use "gamma" (lower cased) to indicate something-north-of-beta. Still, I'm not sure if I'd know what this means as a reader.

Comment (jpignata) [inline docs/lib/auth/fragments/js/advanced.md:11]: "it is relatively safe" doesn't build confidence or give information. Let's stick to the facts: it's not yet marked production, but it is modularized and can reduce the weight of your build size.

**Review by jpignata — CHANGES_REQUESTED** [review]
Some questions about language choices mainly. I'd also like to see some examples of how I'd use these escape hatches, under what scenarios, and some links to learn more (preferably to documentation).

File: docs/lib/auth/fragments/js/advanced.md (hunk 1 above)
Comment (sammartinez) [reply docs/lib/auth/fragments/js/advanced.md:1]: To @jpignata point here, we need to make sure that we are following a similar language or else we need to have legal review this. can we discuss this @harrysolovay and where this is pull request came from?

**Comment by harrysolovay** [comment]
You're absolutely right. This isn't the correct way to address issue #2359 –– thanks for the feedback @jpignata & @sammartinez

**Pull request closed without merging**